flow/flow.go

```diff
@@ -0,0 +1,475 @@
+	f.GrantedAudience = r.GrantedAudience
+	f.ConsentRemember = r.Remember
+	f.ConsentRememberFor = &r.RememberFor
+	f.ConsentHandledAt = r.HandledAt
+	f.ConsentWasHandled = r.WasHandled
+	f.ConsentError = r.Error
+
+	if r.Session != nil {
+		f.SessionIDToken = r.Session.IDToken
+		f.SessionAccessToken = r.Session.AccessToken
+	}
+	return nil
+}
+
+func (f *Flow) InvalidateConsentRequest() error {
+	if f.ConsentWasHandled {
+		return errors.New("consent verifier has already been used")
+	}
+	if f.State != FlowStateConsentUnused && f.State != FlowStateConsentError {
+		return errors.Errorf("unexpected flow state: expected %d or %d, got %d", FlowStateConsentUnused, FlowStateConsentError, f.State)
+	}
+
+	f.ConsentWasHandled = true
+	f.State = FlowStateConsentUsed
+	return nil
+}
+
+func (f *Flow) GetConsentRequest() *consent.ConsentRequest {
+	return &consent.ConsentRequest{
+		ID:                     f.ConsentChallengeID.String(),
+		RequestedScope:         f.RequestedScope,
+		RequestedAudience:      f.RequestedAudience,
+		Skip:                   f.ConsentSkip,
+		Subject:                f.Subject,
+		OpenIDConnectContext:   f.OpenIDConnectContext,
+		Client:                 f.Client,
+		ClientID:               f.ClientID,
+		RequestURL:             f.RequestURL,
+		LoginChallenge:         sqlxx.NullString(f.ID),
+		LoginSessionID:         f.SessionID,
+		ACR:                    f.ACR,
+		AMR:                    f.AMR,
+		Context:                f.Context,
+		WasHandled:             f.ConsentWasHandled,
+		ForceSubjectIdentifier: f.ForceSubjectIdentifier,
+		Verifier:               f.ConsentVerifier.String(),
+		CSRF:                   f.ConsentCSRF.String(),
+		AuthenticatedAt:        f.LoginAuthenticatedAt,
+		RequestedAt:            f.RequestedAt,
+	}
+}
+
+func (f *Flow) GetHandledConsentRequest() *consent.HandledConsentRequest {
+	crf := 0
+	if f.ConsentRememberFor != nil {
+		crf = *f.ConsentRememberFor
+	}
+	return &consent.HandledConsentRequest{
+		ID:                 f.ConsentChallengeID.String(),
+		GrantedScope:       f.GrantedScope,
+		GrantedAudience:    f.GrantedAudience,
+		Session:            &consent.ConsentRequestSessionData{AccessToken: f.SessionAccessToken, IDToken: f.SessionIDToken},
+		Remember:           f.ConsentRemember,
+		RememberFor:        crf,
+		HandledAt:          f.ConsentHandledAt,
+		WasHandled:         f.ConsentWasHandled,
+		ConsentRequest:     f.GetConsentRequest(),
+		Error:              f.ConsentError,
+		RequestedAt:        f.RequestedAt,
+		AuthenticatedAt:    f.LoginAuthenticatedAt,
+		SessionIDToken:     f.SessionIDToken,
+		SessionAccessToken: f.SessionAccessToken,
+	}
+}
+
+func (_ Flow) TableName() string {
+	return "hydra_oauth2_flow"
+}
+
+// FindByConsentChallengeID retrieves a flow given its consent challenge ID.
+func (f *Flow) FindByConsentChallengeID(c *pop.Connection, id string, nid uuid.UUID) error {
+	return c.Where("consent_challenge_id = ? AND nid = ?", id, nid).First(f)
```

Comment:
I've removed this function because it doesn't seem to add any value and we'd have to pass in the persister to use QueryWithNetwork.

Furthermore I'd like to remove `challenge` from the signature of HandleConsentRequest https://github.com/ory/hydra/blob/00100a1bcb60d1836a2c3d6c6a4212e3161b1bda/consent/manager.go#L43, because we require the other argument, `r *HandledConsentRequest` to have the correct challenge id. Passing it in a separate argument is then redundant. Do you agree with this interface change?

Reply:
SGTM